### ISSUE TITLE
Allow Component composition

### ISSUE THREAD

**Issue by renancaraujo** [luanpotter/flame#20]
I have a `SpriteComponent` that will be substituted by an `AnimationComponent` during the game. I've been thinking to compose this logic into a single `PositionComponent` that controls whether I render `SpriteComponent` or `AnimationComponent`, to improve semantics.
Looking into that, how about `Component` class have an `OrderedSet` of subcomponents just like `BaseGame`? I this way, methods like `render` and `update` would be synchronized.

I shall open a pull request in the near future with some ideas in this regard.

**Comment by erickzanardo** [comment]
Wouldn't an new component class be better? Like a `ComposeComponent` or `ComponentGroup` that wraps how many components you need?

This new class would them be responsible by wrapping these inner components and the orchestration of the calls of `render` and `update`.

**Comment by renancaraujo** [comment]
@erickzanardo That coud work. Since my "wrapper" component is a PositionComponent, I think `ComposedComponent` could be a mixin. 
Example of usage:
```dart
class MyComponent extends PositionComponent with ComposedComponent {
...
```

**Comment by luanpotter** [comment]
I like the idea of a mixin, I'm feeling more and more inclined about replacing this hierarchy of components with mixins in the future, I've already created several of them  inside one of my games that could potentially be extracted to flame.

**Comment by luanpotter** [comment]
Merged #25
